examples/gno.land/r/demo/profile/handlers.gno

```diff
@@ -0,0 +1,154 @@
+package profile
+
+import (
+    "bytes"
+    "gno.land/p/demo/ufmt"
+    "gno.land/p/demo/mux"
+    "strings"
+    "std"
+)
+
+const (
+    BaseURL           = "/r/demo/profile"
+    SetStringFieldURL = BaseURL + "?help&__func=SetStringField&field=%s"
+    SetIntFieldURL    = BaseURL + "?help&__func=SetIntField&field=%s"
+    SetBoolFieldURL   = BaseURL + "?help&__func=SetBoolField&field=%s"
+    GetStringFieldURL = BaseURL + "?help&__func=GetStringField&addr=%s"
+    GetIntFieldURL    = BaseURL + "?help&__func=GetIntField&addr=%s"
+    GetBoolFieldURL   = BaseURL + "?help&__func=GetBoolField&addr=%s"
+    ViewAllFieldsURL  = BaseURL + ":u/%s"
+    ViewFieldURL      = BaseURL + ":f/%s/%s"
+)
+
+func homeHandler(res *mux.ResponseWriter, req *mux.Request) {
+    var b bytes.Buffer
+    addresses := make(map[string]struct{})
+
+    b.WriteString("<html><body>")
+    b.WriteString("<h1>Profiles</h1>")
+    b.WriteString("<h2>Setters</h2>")
+    for field := range stringFields {
+        link := ufmt.Sprintf(SetStringFieldURL, field)
+        b.WriteString(ufmt.Sprintf("<p>[<a href='%s'>Set %s Field</a>]</p>", link, field))
+    }
+
+    for field := range intFields {
+        link := ufmt.Sprintf(SetIntFieldURL, field)
+        b.WriteString(ufmt.Sprintf("<p>[<a href='%s'>Set %s Field</a>]</p>", link, field))
+    }
+
+    for field := range boolFields {
+        link := ufmt.Sprintf(SetBoolFieldURL, field)
+        b.WriteString(ufmt.Sprintf("<p>[<a href='%s'>Set %s Field</a>]</p>", link, field))
+    }
+
+    b.WriteString("<p>--------------------------</p>")
+
+    fields.Iterate("", "", func(key string, value interface{}) bool {
+        parts := strings.SplitN(key, ":", 2)
+        if len(parts) > 0 {
+            addresses[parts[0]] = struct{}{}
+        }
+        return true
+    })
+
+    b.WriteString("<h2>All Address Profiles</h2>")
+
+    for address := range addresses {
+        getStringLink := ufmt.Sprintf(GetStringFieldURL, address)
+        getIntLink := ufmt.Sprintf(GetIntFieldURL, address)
+        getBoolLink := ufmt.Sprintf(GetBoolFieldURL, address)
+
+        b.WriteString(ufmt.Sprintf(
+            "<p>Address: %s </br>"+
+                "[<a href='%s'>View All Fields</a>] "+
+                "[<a href='%s'>Get String Field</a>] "+
+                "[<a href='%s'>Get Int Field</a>] "+
+                "[<a href='%s'>Get Bool Field</a>]</p>",
```

Comment:
we should not promote using a transaction to "Get" a field, we should instead make the "Render()" displaying them.

Reply:
https://github.com/gnolang/gno/pull/1983/commits/6755ac92c64d1ba544ba57fc71bd4c58a81799bd